### ISSUE TITLE
Docs: Update the AbstractAuthenticationMiddleware documentation

### ISSUE THREAD

**Issue by ctrl-Felix** [litestar-org/litestar#1814]
### Summary

Link: https://docs.litestar.dev/2/usage/security/abstract-authentication-middleware.html

The part which explains how to access the User and the Token from the object is wrong:
```python
from litestar import Request, get

from my_app.db.models import User
from my_app.security.jwt import Token


@get("/")
def my_route_handler(request: Request[User, Token]) -> None:
    user = request.user  # correctly typed as User
    auth = request.auth  # correctly typed as Token
    assert isinstance(user, User)
    assert isinstance(auth, Token)
```

Object reference: https://docs.litestar.dev/2/reference/connection.html#litestar.connection.Request

I assume that the Webocket part is wrong too:

```python
from litestar import WebSocket, websocket

from my_app.db.models import User
from my_app.security.jwt import Token


@websocket("/")
async def my_route_handler(socket: WebSocket[User, Token]) -> None:
    user = socket.user  # correctly typed as User
    auth = socket.auth  # correctly typed as Token
    assert isinstance(user, User)
    assert isinstance(auth, Token)
```

**Comment by cofin** [comment]
@allcontributors please add @ctrl-Felix for bug reports

**Comment by provinzkraut** [comment]
Fixed in #1815.
